block/gs/adapter.go

```diff
@@ -121,6 +121,29 @@ func (a *Adapter) Get(obj block.ObjectPointer, _ int64) (io.ReadCloser, error) {
 	return r, err
 }
 
+func (a *Adapter) List(storageNamespace, prefix string) ([]string, error) {
+	var err error
+	defer reportMetrics("List", time.Now(), nil, &err)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var keys []string
+	iter := a.client.Bucket(storageNamespace).Objects(context.Background(), &storage.Query{Prefix: prefix})
```

Comment:
Missing namespace resolve.
Every adapter today (which I think we can consolidate) resolves the object path using the given namespace and turn it into the actual storage namespace (aka bucket) - namespace in overloaded.
If you use the repository namespace directly it will not work as it will include the bucket name with an optional path, which in this case will fail.

Reply:
I see, good catch. Fixed.